src/sentry/replays/tasks.py

```diff
@@ -20,16 +22,22 @@
 )
 def delete_recording_segments(project_id: int, replay_id: str, **kwargs: dict) -> None:
     """Asynchronously delete a replay."""
-    archive_replay(project_id, replay_id)
     delete_replay_recording(project_id, replay_id)
+    archive_replay(project_id, replay_id)
 
 
 def delete_replay_recording(project_id: int, replay_id: str) -> None:
     """Delete all recording-segments associated with a Replay."""
-    segments = ReplayRecordingSegment.objects.filter(
+    segments = fetch_segments_metadata(project_id, replay_id, offset=0, limit=10000)
+    for segment in segments:
+        driver = FilestoreBlob() if segment.file_id else StorageBlob()
+        driver.delete(segment)
+
+    # delete old rows from SQL too
+    segment_sql_rows = ReplayRecordingSegment.objects.filter(
         replay_id=replay_id, project_id=project_id
     ).all()
-    for segment in segments:
+    for segment in segment_sql_rows:
         segment.delete()  # Three queries + one request to the message broker
```

Comment:
We should update `ReplayRecordingSegment.delete` to not query for the File model since this is being handled by the `Filestore.delete` method.

Reply:
I think how it's currently written is fine -- it will skip over the file relations and go ahead and dleete the row https://github.com/getsentry/sentry/blob/a9434658c958646c05df020f286a59d708c02a5f/src/sentry/replays/models.py#L49